### ISSUE TITLE
Errors on startup...

### ISSUE THREAD

**Issue by JanSchulz** [ipython-contrib/IPython-notebook-extensions#490]
```
[W 20:50:25.063 NotebookApp] Collisions detected in jupyter_notebook_config.py a
nd jupyter_notebook_config.json config files. jupyter_notebook_config.json has h
igher priority: {
      "Exporter": {
        "template_path": "['.', 'C:\\\\Users\\\\jschulz\\\\AppData\\\\Roaming\\\
\jupyter\\\\templates'] ignored, using ['C:\\\\Users\\\\jschulz\\\\AppData\\\\Ro
aming\\\\jupyter\\\\templates']"
      }
    }
```
Seems that I should remove something from my old config?

```pytb
[W 20:50:25.148 NotebookApp] Error loading server extension highlight_latex_cfg
    Traceback (most recent call last):
      File "C:\portabel\miniconda\envs\ipython-notebook\lib\site-packages\notebo
ok\notebookapp.py", line 1005, in init_server_extensions
        mod = importlib.import_module(modulename)
      File "C:\portabel\miniconda\envs\ipython-notebook\lib\importlib\__init__.p
y", line 109, in import_module
        return _bootstrap._gcd_import(name[level:], package, level)
      File "<frozen importlib._bootstrap>", line 2254, in _gcd_import
      File "<frozen importlib._bootstrap>", line 2237, in _find_and_load
      File "<frozen importlib._bootstrap>", line 2226, in _find_and_load_unlocked
      File "<frozen importlib._bootstrap>", line 1200, in _load_unlocked
      File "<frozen importlib._bootstrap>", line 1129, in _exec
      File "<frozen importlib._bootstrap>", line 1471, in exec_module
      File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
      File "C:\Users\jschulz\AppData\Roaming\jupyter\extensions\highlight_latex_cfg.py", line 7, in <module>
        c = get_config()
    NameError: name 'get_config' is not defined
[W 20:50:25.316 NotebookApp] Error loading server extension highlight_html_cfg
    Traceback (most recent call last):
      File "C:\portabel\miniconda\envs\ipython-notebook\lib\site-packages\notebook\notebookapp.py", line 1005, in init_server_extensions
        mod = importlib.import_module(modulename)
      File "C:\portabel\miniconda\envs\ipython-notebook\lib\importlib\__init__.py", line 109, in import_module
        return _bootstrap._gcd_import(name[level:], package, level)
      File "<frozen importlib._bootstrap>", line 2254, in _gcd_import
      File "<frozen importlib._bootstrap>", line 2237, in _find_and_load
      File "<frozen importlib._bootstrap>", line 2226, in _find_and_load_unlocked
      File "<frozen importlib._bootstrap>", line 1200, in _load_unlocked
      File "<frozen importlib._bootstrap>", line 1129, in _exec
      File "<frozen importlib._bootstrap>", line 1471, in exec_module
      File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
      File "C:\Users\jschulz\AppData\Roaming\jupyter\extensions\highlight_html_cfg.py", line 7, in <module>
        c = get_config()
    NameError: name 'get_config' is not defined
```
no idea :-(

```pytb
[W 20:50:25.361 NotebookApp] Error loading server extension ipy-aspell-server
    Traceback (most recent call last):
      File "C:\portabel\miniconda\envs\ipython-notebook\lib\site-packages\notebo
ok\notebookapp.py", line 1005, in init_server_extensions
        mod = importlib.import_module(modulename)
      File "C:\portabel\miniconda\envs\ipython-notebook\lib\importlib\__init__.p
y", line 109, in import_module
        return _bootstrap._gcd_import(name[level:], package, level)
      File "<frozen importlib._bootstrap>", line 2254, in _gcd_import
      File "<frozen importlib._bootstrap>", line 2237, in _find_and_load
      File "<frozen importlib._bootstrap>", line 2226, in _find_and_load_unlocked
      File "<frozen importlib._bootstrap>", line 1200, in _load_unlocked
      File "<frozen importlib._bootstrap>", line 1129, in _exec
      File "<frozen importlib._bootstrap>", line 1467, in exec_module
      File "<frozen importlib._bootstrap>", line 1572, in get_code
      File "<frozen importlib._bootstrap>", line 1532, in source_to_code
      File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
      File "C:\Users\jschulz\AppData\Roaming\jupyter\extensions\ipy-aspell-server.py", line 40
        print "open socket"
                          ^
    SyntaxError: Missing parentheses in call to 'print'
```

yep, my notebook server runs in py34 :-)

**Comment by JanSchulz** [comment]
The print issue is https://github.com/ipython-contrib/IPython-notebook-extensions/pull/491

**Comment by jcb91** [comment]
So for the first one, there's reference on the wiki page about something to do with clusters: https://github.com/ipython-contrib/IPython-notebook-extensions/wiki/Home-4.x-(Jupyter)#configuration

For the second, I'd recommend not enabling the `ipy-aspell-server` server extension at all, unless you really are using it. As noted in its readme (I think? Can't remember off the top of my head), it isn't terribly secure, and is redundant given js-only spell checkers.

For the second, I'm not sure...

**Comment by halflings** [comment]
Just confirming that I have the same errors.

That "get_config" is not defined anywhere, so I'm not sure why it's called there.

**Comment by jcb91** [comment]
The `get_config` is a standard call in configuration scripts (see e.g. [jupyter_notebook_config.py#L7](https://github.com/ipython-contrib/IPython-notebook-extensions/blob/f9549521d6bac315baf094649ead1f9a93344ec3/jupyter_notebook_config.py#L7), [nbextensions/usability/highlighter/nbconvert_config.py#L7](https://github.com/ipython-contrib/IPython-notebook-extensions/blob/0b1c8af2094bc8e5cd8fe425f81fe9b7f8d232d0/nbextensions/usability/highlighter/nbconvert_config.py#L7), [configure_nbextensions.py#L45](https://github.com/ipython-contrib/IPython-notebook-extensions/blob/079686f4db83289152aa2b5d173822062c6980fe/configure_nbextensions.py#L45) etc). So, the real question is why `highlight_latex_cfg.py` is being loaded as a server extension at all, since it isn't one :confused:

**Comment by jcb91** [comment]
aaaand the reason for that is that they're left in the `/extensions` directory of the repo, and since #476, we add all `.py` files in that dir to be loaded as extensions. They should probably be removed from the repo altogether, or merged into the main configure script.

**Comment by jcb91** [comment]
actually it seems the purpose of `highlight_latex_cfg.py` and `highlight_latex_cfg.py` is to switch between two distinct output formats, so they can't be merged into the main config. At the same time, they can't stay in `/extensions` if we keep #476 as-is. @jfbercher,any thoughts?

**Comment by jcb91** [comment]
So the second two should be fixed by #499, did you manage to resolve the first one, about the config conflict, @JanSchulz ?

**Comment by jfbercher** [comment]
@jcb91, sorry I missed this thread and I am late to the party. Indeed, the
extension directory also contains non server scripts, which may be useful
since this directory is added to the path of any user. Perhaps should we
devote another directory for this purpose (or a specific directory for
server extensions).

**Comment by jcb91** [comment]
No worries @jfbercher. Since #499 reverted #476, this is not so much of a pressing issue anymore. I think manual addition of extensions to enable in the config is probably safer and simpler than trying to add lots of extra directories to the path, at least for now.

**Comment by juhasch** [comment]
Should be fixed by #499

**Comment by jaypalsing** [comment]
its fix?
